python/src/aiconfig/callback.py

```diff
@@ -0,0 +1,139 @@
+# Standard Libraries
+import asyncio
+import json
+import time
+
+# Third Party Libraries
+from result import Err, Ok
+from pydantic import BaseModel, ConfigDict
+from typing import (
+    Coroutine,
+    Final,
+    List,
+    Sequence,
+    TextIO,
+    TypeAlias,
+    Union,
+    Any,
+    Awaitable,
+    Callable,
+    Optional,
+)
+
+# Constants
+DEFAULT_TIMEOUT = 5  # Default timeout for callback execution in seconds
+
+
+class CallbackEvent:
+    """
+    Represents an event with data to be passed to a callback.
+    """
+
+    def __init__(self, name: str, data: Any, ts_ns: int = time.time_ns()):
+        self.name = name
+        # Anything available at the time the event happens.
+        # It is passed to the callback.
+        self.data = data
+        self.ts_ns = ts_ns
+
+
+# Type Aliases
+Callback = Callable[[CallbackEvent], Awaitable]
+Result: TypeAlias = Union[Ok[Any], Err[Any]]
```

Comment:
This is defined already in the Result library. Also, let's try to parametrize with generics where possible and avoid any

Reply:
What's the reason to avoid Any?

Reply:
```
# static type error (good, we want the error)
def f(x: str) -> int:
    return x

def g(y: int): -> int
  return y + 1

g(f("3"))

# no static type error (bad, we dont wan't to have a runtime error)
def f(x: str) -> Any:
    return 1

# TypeError: cannot add "3" to 1.
g(f("3"))
```

Similar idea with generic type arguments.